syntax/parse.rs

```diff
@@ -1281,6 +1282,20 @@ fn parse_type_path(ty: &TypePath) -> Result<Type> {
                             rangle: generic.gt_token,
                         })));
                     }
+                } else if ident == "Option" {
+                    if generic.args.len() == 1 {
+                        if let GenericArgument::Type(arg) = &generic.args[0] {
+                            let inner = parse_type(arg)?;
+                            return Ok(Type::RustOption(Box::new(Ty1 {
+                                name: ident,
+                                langle: generic.lt_token,
+                                inner,
+                                rangle: generic.gt_token,
+                            })));
+                        }
+                    } else {
+                        panic!("{}", generic.args.len());
+                    }
```

Comment:
Panic here seems inappropriate, and inconsistent with the patten employed by the rest of the types.
```suggestion
                } else if ident == "Option" && generic.args.len() == 1 {
                    if let GenericArgument::Type(arg) = &generic.args[0] {
                        let inner = parse_type(arg)?;
                        return Ok(Type::RustOption(Box::new(Ty1 {
                            name: ident,
                            langle: generic.lt_token,
                            inner,
                            rangle: generic.gt_token,
                        })));
                    }
```